arroyo/processing/processor.py

```diff
@@ -79,21 +79,24 @@ def _create_strategy() -> None:
 
         def on_partitions_assigned(partitions: Mapping[Partition, int]) -> None:
             logger.info("New partitions assigned: %r", partitions)
-            if self.__processing_strategy is not None:
-                _close_strategy()
-            _create_strategy()
+            if partitions:
```

Comment:
A hypothetical scenario.

Let's say you get new partitions assigned while the consumer is shutting down. That means you have set shutdown_requested to True over here https://github.com/getsentry/arroyo/blob/main/arroyo/processing/processor.py#L223. Is there still a need to keep creating the strategy again? Or can we check for the flag and do nothing in those scenarios?

Reply:
I spent some time deliberating about this. I decided not to add it because it shouldn't be necessary. During shutdown all partitions should have be revoked and thus the strategy will not be recreated anyway (since `self.__consumer.tell().keys() - set(partitions)` should be falsy.